src/util/validateRoam.ts

```diff
@@ -0,0 +1,32 @@
+import { RoamBlock, RoamPage } from 'roam'
+
+/**
+ * Validates the strucutre of RoamBlocks.
+ */
+const isRoamBlock = (children: RoamBlock[] = []): boolean => children.every(({
+  uid,
+  string,
+  children = [],
+  'create-time': createTime,
+  'create-email': createEmail
+}: RoamBlock) =>
+  Array.isArray(children)
+  && isRoamBlock(children)
+  && typeof uid === 'string'
+  && typeof string === 'string'
+  && typeof createTime === 'number'
+  && typeof createEmail === 'string'
+)
```

Comment:
I think it's generally cleaner to extract the mapping logic outside of functions. So `isRoamBlock` would just take a single `RoamBlock`. The aggregate can be calculated with `children.every(isRoamBlock)` in `validateRoam`. This decouples the validation from the aggregate structure.